### ISSUE TITLE
NBT Explorer Integration

### ISSUE THREAD

**Issue by mhGandhi** [PrismLauncher/PrismLauncher#3087]
### Role

I sometimes need to modify or view NBT data of worlds or players within without loading the world. (e.g. when getting kicked by a broken modded item)

### Suggestion

Add a button on the worlds screen to edit the worlds NBT-Data

### Benefit

Users can edit their worlds without loading them, and they don't have to navigate to the path in NBT Explorer

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

There is already a program that does that (https://github.com/jaquadro/NBTExplorer). It would probably suffice to just have a button in the worlds-screen, that opens NBT-Explorer with the correct path.

**Comment by TheKodeToad** [comment]
If this happens it'll probably just be a button to open NBT explorer - a full NBT viewer and editor is not really something which belongs in the launcher.

Unfortunately, NBT explorer does not integrate well with Linux :(

I personally was trying to work on a nice cross platform editor but did not get too far with that

**Comment by mhGandhi** [comment]
> If this happens it'll probably just be a button to open NBT explorer - a full NBT viewer and editor is not really something which belongs in the launcher.
> 
> Unfortunately, NBT explorer does not integrate well with Linux :(
> 
> I personally was trying to work on a nice cross platform editor but did not get too far with that

Yes, that is what I wrote, a button that opens the program with the corresponding path.
I didn't know that there were compatibility issues between Linux and Prism so for now ill just open the path manually.

**Comment by TheKodeToad** [comment]
No... we do try to support Linux as well as possible... but NbtExplorer is a C# program that only really works properly on Windows and Mac - on Linux it is ugly, does not open .minecraft correctly and has an awkward to use file browser... but I suppose that is most users 

Correction: The Mac version is unusable as of macOS Catalina due to 32-bit support being removed from the operating system.

**Comment by TheKodeToad** [comment]
Hmm... actually it works well enough... I was probably being a bit nitpicky. You don't actually need the native Mac version either, you can just run it the same way you do on Linux - using mono. (would not work on flatpak though I don't think!)

There also exists NBT Studio which has more features and is more up to date. Maybe we should support that instead...

**Comment by mhGandhi** [comment]
> Hmm... actually it works well enough... I was probably being a bit nitpicky. You don't actually need the native Mac version either, you can just run it the same way you do on Linux - using mono. (would not work on flatpak though I don't think!)
> 
> There also exists NBT Studio which has more features and is more up to date. Maybe we should support that instead...

Yeah I checked it out briefly rn. It is supposed to be "the spiritual successor" so I guess it would make sense to support that one rather than NBT-Explorer as it does seem to have some additional features.

**Comment by ludgie** [comment]
There is jdNBTExplorer for Linux & Windows. It is available as a flatpak through Flathub too. Written with python and Qt 6.
https://codeberg.org/JakobDev/jdNBTExplorer

Support could be added for this tool and some other tool that provides macOS (Intel & arm) support.
